storage/queue.go

```diff
@@ -170,9 +179,26 @@ func (bq *baseQueue) Add(repl *Replica, priority float64) error {
 // not be added, as the replica with the lowest priority will be
 // dropped.
 func (bq *baseQueue) MaybeAdd(repl *Replica, now proto.Timestamp) {
+	// Load the system config.
+	cfg, err := bq.gossip.GetSystemConfig()
```

Comment:
Eek, do we always want to load the system config on every pass? This thing is going to get fairly large and that's a lot of proto decoding. Why not set up a gossip callback on the system config key and cache the value of `GetSystemConfig()` when the callback fires?